### ISSUE TITLE
fix: data leakage — preprocessing pipeline fitted before train/test split

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#929]
## Bug Description

The preprocessing pipeline is fitted on the **entire dataset** before the train/test/validation split, causing data leakage. The scaler (e.g., `StandardScaler`) learns the mean and standard deviation from ALL data including what becomes the test set, leading to overly optimistic validation and test metrics.

## Root Cause

In `AiModelBuilder.BuildSupervisedInternalAsync()`, the preprocessing pipeline is fitted first, then the data is split:

**`src/AiModelBuilder.cs`**
```csharp
// Line 2043: FitTransform on ALL data (WRONG — sees test data)
preprocessedX = _preprocessingPipeline.FitTransform(preparedX);
preprocessedY = preparedY;

// ... 58 lines later ...

// Line 2101: Split AFTER fitting (TOO LATE — pipeline already contaminated)
(XTrain, yTrain, XVal, yVal, XTest, yTest) = DataSplitter.Split<T, TInput, TOutput>(
    preprocessedX, preprocessedY, trainRatio: 0.7, validationRatio: 0.15, shuffle: true);
```

## Expected Behavior

The correct ML pipeline ordering is:
1. Split data into train/validation/test
2. Fit preprocessing ONLY on training data
3. Transform validation and test using the fitted-on-training pipeline

```csharp
// CORRECT ordering:
(XTrain, yTrain, XVal, yVal, XTest, yTest) = DataSplitter.Split<T, TInput, TOutput>(
    preparedX, preparedY, trainRatio: 0.7, validationRatio: 0.15, shuffle: true);

// Fit ONLY on training data
var preprocessedXTrain = _preprocessingPipeline.FitTransform(XTrain);

// Transform (not FitTransform) validation and test sets
var preprocessedXVal = _preprocessingPipeline.Transform(XVal);
var preprocessedXTest = _preprocessingPipeline.Transform(XTest);
```

## Impact

- **Validation and test metrics are inflated** — they reflect knowledge the model shouldn't have
- Affects ALL models built through `AiModelBuilder` with preprocessing enabled
- Users relying on the reported metrics for model selection may choose overfit models
- Production performance will be worse than the reported metrics suggest

## Files to Modify

| File | Line | Change |
|------|------|--------|
| `src/AiModelBuilder.cs` | 2043-2103 | Move `DataSplitter.Split()` BEFORE `FitTransform()`. Fit on XTrain only. Transform XVal/XTest separately. |

## Severity

**Medium** — Does not crash, but produces misleading evaluation metrics that can cause poor model selection decisions in production.

## Notes

- The `PreprocessingInfo` stored on `AiModelResult` already holds a reference to the fitted pipeline, so `Predict()` on new data will correctly use the fitted pipeline. The issue is only that the fitted parameters include test data statistics.
- The federated learning path at line 2091 (`usePartitionedFederatedData`) correctly uses ALL data as training, so it is NOT affected by this bug.
- The `DataSplitter.Split()` uses `shuffle: true` with a fixed seed, so the split is deterministic.

Related: #927
